x/swaprouter/creator.go

```diff
@@ -110,17 +120,21 @@ func (k Keeper) validateCreatedPool(
 		return errors.Wrapf(types.ErrInvalidPool,
 			"Pool was attempted to be created with incorrect pool address.")
 	}
-	// Notably we use the initial pool liquidity at the start of the messages definition
-	// just in case CreatePool was mutative.
-	if !pool.GetTotalPoolLiquidity(ctx).IsEqual(initialPoolLiquidity) {
-		return errors.Wrap(types.ErrInvalidPool,
-			"Pool was attempted to be created, with initial liquidity not equal to what was specified.")
-	}
-	// TODO: this check should be moved
-	// This check can be removed later, and replaced with a minimum.
-	if !pool.GetTotalShares().Equal(gammtypes.InitPoolSharesSupply) {
-		return errors.Wrap(types.ErrInvalidPool,
-			"Pool was attempted to be created with incorrect number of initial shares.")
+
+	// Check the total pool liquidity/shares if pool is not a concentrated liquidity pool
+	if poolType != types.Concentrated {
+		// Notably we use the initial pool liquidity at the start of the messages definition
+		// just in case CreatePool was mutative.
+		if !pool.GetTotalPoolLiquidity(ctx).IsEqual(initialPoolLiquidity) {
+			return errors.Wrap(types.ErrInvalidPool,
+				"Pool was attempted to be created, with initial liquidity not equal to what was specified.")
+		}
+		// TODO: this check should be moved
+		// This check can be removed later, and replaced with a minimum.
+		if !pool.GetTotalShares().Equal(gammtypes.InitPoolSharesSupply) {
+			return errors.Wrap(types.ErrInvalidPool,
+				"Pool was attempted to be created with incorrect number of initial shares.")
+		}
```

Comment:
Can we move this into `gamm` under `validateCFMMPool` method and call it right at the beginning of `InitializePool`?

Reply:
I dont think so, because we would need to pass initialPoolLiquidity into it somehow

Reply:
But then we would be checking if pool.GetTotalLiquidity() isEqual to pool.GetTotalLiquidity() 😅

Reply:
Ah I see. In that case, this check seems unnecessary. Validation such as this (checking that pool was initialized with correct liquidity and shares) should be done by unit tests, not at runtime.

We might want to make sure that the unit tests exist and remove these